viewing_party/main.py

```diff
@@ -0,0 +1,172 @@
+
+###################### WAVE 1 PASSED #########################
+### Wave 1.1 (4 tests - PASSED) ###
+def create_movie(movie_title, genre, rating):
+    
+    new_movie = {
+        "title": movie_title,
+        "genre": genre,
+        "rating": rating,
+    }       
+    
+    if new_movie["title"] == None:
+        new_movie = None
+    elif new_movie["genre"] == None:
+        new_movie = None
+    elif new_movie["rating"] == None:
+        new_movie = None
+    return new_movie
+
+### Wave 1.2 (1 test PASSED) ###
+def add_to_watched(user_data, movie):    
+    # user_data is a dictionary key"watched": value lists title,genre,rating
+    # updated_data is a list of dictionaries
+    user_data["watched"].append(movie)
+    return user_data
+
+### Wave 1.3 (1 test PASSED) ###
+def add_to_watchlist(user_data, movie): 
+    user_data["watchlist"].append(movie)
+    return user_data
+
+### Wave 1.4 (3 tests PASSED) ###
+def watch_movie(user_data, title):
+    for i in user_data["watchlist"]:
+        if i["title"] == title:
+            user_data["watchlist"].remove(i)
+            user_data["watched"].append(i)
+    return user_data
+
+
+###################### WAVE 2 PASSED #########################
+### Wave 2.1 (2 tests) PASSED ###
+def get_watched_avg_rating(user_data):
+    watched_count = 0
+    rating_sum = 0.0
+    average = 0.0
+    if user_data["watched"] == []:
+        return average
+    else:
+        for i in user_data["watched"]:
+            watched_count += 1
+            rating_sum += i["rating"]
+    average = rating_sum / watched_count
+    return average
+
+### Wave 2.2 (2 tests) ###
+def get_most_watched_genre(user_data):
+    popular_genre = " "
+    user_genres = {}
+    genre_sort = []
+
+    if user_data["watched"] == []:
+        popular_genre = None
+        return popular_genre
+    for i in user_data["watched"]:
+        if i["genre"] in user_genres:
+            user_genres[i["genre"]] += 1
+        else:
+            user_genres[i["genre"]] = 1
+        # user_genres = {"Fanstasy": 2, "Intrigue": 3}
+    genre_sort = sorted(user_genres)
+    # sorted puts user_genres' keys in a list ordered from lowest to highest value
+    # ...so ... genre_sort = ["Fantasy", "Intrigue"] 
+    popular_genre = genre_sort[-1]
```

Comment:
This works, but it *is* a bit slower to sort a collection than simply to find the max value.  For this problem this absolutely works, this note is just an FYI.